### PR TITLE
tool to enumerate charges using dimorphite

### PR DESCRIPTION
**Pull request by tdudgeon**
simple tool that enumerates charge states of molecules

### CODE REVIEW

**Comment by simonbray** [comment]
Hm. It seems the CI testing is using Galaxy release 19.05 which doesn't have the `has_n_lines` assertion: https://docs.galaxyproject.org/en/release_19.05/dev/schema.html

File: .travis.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
 
 before_install:
   - export GALAXY_REPO=https://github.com/galaxyproject/galaxy
-  - export GALAXY_RELEASE=release_19.05
+  - export GALAXY_RELEASE=release_20.05
```
Comment (simonbray) [inline .travis.yml]: hope this is okay Björn

**Comment by simonbray** [comment]
would be nice to have a slightly more robust test I suppose, but I see from Frank's mail you want to get this merged quickly

**Comment by simonbray** [comment]
@bgruening any objection if we merge

File: chemicaltoolbox/rdkit/enumerate_charges.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <tests>
+        <test>
+            <param name="input" ftype='smi' value="mols.smi" />
+            <param name="min_ph"  value="6.4" />
+            <param name="max_ph"  value="8.4" />
+            <output name="output" ftype='smi'>
+                <!-- there are problems comparing the output to the expected due to whitespace differences
+                so we just check that the number of lines is as expected -->
+                <assert_contents>
+                    <has_n_lines n="7" />
```
Comment (bgruening) [inline chemicaltoolbox/rdkit/enumerate_charges.xml]: @tdudgeon can you add here a few more asserts ... checking for a few smiles that are expected etc?

File: chemicaltoolbox/rdkit/enumerate_charges.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                </assert_contents>
+            </output>
+        </test>
+        <test>
+            <param name="input" ftype='smi' value="mols.smi" />
+            <param name="min_ph"  value="4.4" />
+            <param name="max_ph"  value="10.4" />
+            <output name="output" ftype='smi'>
+                <assert_contents>
+                    <has_n_lines n="14" />
```
Comment (bgruening) [inline chemicaltoolbox/rdkit/enumerate_charges.xml]: same here, a few more asserts would be great

**Comment by simonbray** [comment]
@bgruening, we want to use this tool in a workflow, is it ok if we just merge?

I can add a proper test later.

**Comment by simonbray** [comment]
Tests are passing, will merge.

**Pull request merged**